### PR TITLE
fix(cli/trigger): disallow empty connection on create

### PR DESCRIPTION
**Pull request by ig-ra**
- ability to threat empty args as not-provided ones
- report not found types in errors
- disallow empty connection in trigger create

### CODE REVIEW

File: cmd/ak/common/flags.go [hunk 1]
```diff
@@ -68,3 +69,19 @@ func ToExitCodeWithSkipNotFoundFlag(cmd *cobra.Command, err error, whats ...stri
 	}
 	return exitErr
 }
+
+// maybe we could just run this check for all required flags instead of providing explicit ones?
```
Comment (itayd) [inline cmd/ak/common/flags.go:73]: give it a shot

File: cmd/ak/common/flags.go [hunk 2]
```diff
@@ -68,3 +69,19 @@ func ToExitCodeWithSkipNotFoundFlag(cmd *cobra.Command, err error, whats ...stri
...
+func UnsetEmptyFlags(cmd *cobra.Command, flags ...string) {
+	flagsMap := make(map[string]struct{}, len(flags))
+	for _, flag := range flags {
+		flagsMap[flag] = struct{}{}
+	}
+
```
Comment (itayd) [inline cmd/ak/common/flags.go:79]: why is this needed?

File: cmd/ak/common/flags.go [hunk 3]
```diff
@@ -68,3 +69,19 @@ func ToExitCodeWithSkipNotFoundFlag(cmd *cobra.Command, err error, whats ...stri
 	}
 	return exitErr
 }
+
+// maybe we could just run this check for all required flags instead of providing explicit ones?
+func UnsetEmptyFlags(cmd *cobra.Command, flags ...string) {
+	flagsMap := make(map[string]struct{}, len(flags))
```
Comment (itayd) [inline cmd/ak/common/flags.go:75]: it's easier just to `make(map[string]bool, len(flags)`. then check if `if flagsMap[whatever] {`.

File: cmd/ak/common/flags.go [hunk 4]
```diff
@@ -68,3 +69,19 @@ func ToExitCodeWithSkipNotFoundFlag(cmd *cobra.Command, err error, whats ...stri
...
+	for _, flag := range flags {
+		flagsMap[flag] = struct{}{}
+	}
+
+	cmd.Flags().Visit(func(flag *pflag.Flag) {
+		if _, ok := flagsMap[flag.Name]; ok {
+			if flag.Value.Type() == "string" && flag.Changed && flag.Value.String() == "" {
+				flag.Changed = false
```
Comment (itayd) [inline cmd/ak/common/flags.go:83]: don't we want to reset the value as well?

File: cmd/ak/common/flags.go (hunk 1 above)
Comment (ig-ra) [reply cmd/ak/common/flags.go:73]: I'm afraid to apply this to all required arguments actually :) That's why 
So I prefer (meanwhile) to explicitly specify commands and which arguments cannot be empty

File: cmd/ak/common/flags.go (hunk 3 above)
Comment (ig-ra) [reply cmd/ak/common/flags.go:75]: sorry, I failed to understand your comment.
The idea is to pass in `flags` arguments that should be unset if they are empty. This will allow cobra to run all other validation logic as if they were not passed by the user.
Therefore I want to convert them to map to easily check later in callback called from `Visit` function on each arg set by user

File: cmd/ak/common/flags.go (hunk 2 above)
Comment (ig-ra) [reply cmd/ak/common/flags.go:79]: I don't want to unset all empty args passed by user, but only those I asked explicitly in `flags`. So in `Visit` which will visit all user-set args I need to check whether the args is one from the list requested

File: cmd/ak/common/flags.go (hunk 4 above)
Comment (ig-ra) [reply cmd/ak/common/flags.go:83]: no, because the value in both cases is `""` e.g. empty string.
But in one it's a default value for argument and in the other it's the value passed by the user
